indexing-service/src/test/java/io/druid/indexing/common/task/HadoopTaskTest.java

```diff
@@ -0,0 +1,104 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.common.task;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.druid.indexer.updater.HadoopDruidConverterConfig;
+import io.druid.indexing.common.TaskStatus;
+import io.druid.indexing.common.TaskToolbox;
+import io.druid.indexing.common.actions.TaskActionClient;
+import io.druid.indexing.common.config.TaskConfig;
+import org.apache.hadoop.yarn.util.ApplicationClassLoader;
+import org.easymock.EasyMock;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.net.URLClassLoader;
+
+public class HadoopTaskTest
+{
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Test
+  public void testBuildClassLoader() throws Exception
+  {
+    final HadoopTask task = new HadoopTask(
+        "taskId",
+        "dataSource",
+        ImmutableList.<String>of(),
+        ImmutableMap.<String, Object>of()
+    )
+    {
+      @Override
+      public String getType()
+      {
+        return null;
+      }
+
+      @Override
+      public boolean isReady(TaskActionClient taskActionClient) throws Exception
+      {
+        return false;
+      }
+
+      @Override
+      public TaskStatus run(TaskToolbox toolbox) throws Exception
+      {
+        return null;
+      }
+    };
+    final TaskToolbox toolbox = EasyMock.createStrictMock(TaskToolbox.class);
+    EasyMock.expect(toolbox.getConfig()).andReturn(new TaskConfig(
+        temporaryFolder.newFolder().toString(),
+        null,
+        null,
+        null,
+        ImmutableList.of("something:hadoop:1"),
+        false,
+        null,
+        null
+    )).once();
+    EasyMock.replay(toolbox);
+
+    final ClassLoader classLoader = task.buildClassLoader(toolbox);
+    checkDoesNotGoToApplicationClassloader(classLoader);
+
+    Assert.assertTrue(classLoader instanceof URLClassLoader);
+    Assert.assertFalse(classLoader instanceof ApplicationClassLoader);
+
+    final Class<?> hadoopClazz = Class.forName("org.apache.hadoop.fs.FSDataInputStream", false, classLoader);
+    checkDoesNotGoToApplicationClassloader(hadoopClazz.getClassLoader());
+
+    final Class<?> druidHadoopConfigClazz = Class.forName("io.druid.indexer.HadoopDruidIndexerConfig", false, classLoader);
+    checkDoesNotGoToApplicationClassloader(druidHadoopConfigClazz.getClassLoader());
+  }
+  public static void checkDoesNotGoToApplicationClassloader(ClassLoader classLoader)
+  {
+    final ClassLoader appLoader = HadoopDruidConverterConfig.class.getClassLoader();
+    for(ClassLoader parent = classLoader; parent != null; parent=parent.getParent()){
```

Comment:
but, if you have classloaders in a chain then things fall apart and #2500 ensures that there is no chain and test should be enforcing same.